presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoPageSource.java

```diff
@@ -53,10 +55,11 @@
 import static com.google.common.base.Verify.verify;
 import static io.airlift.slice.Slices.utf8Slice;
 import static io.airlift.slice.Slices.wrappedBuffer;
+import static java.util.concurrent.CompletableFuture.completedFuture;
 import static java.util.stream.Collectors.toList;
 
 public class MongoPageSource
-        implements ConnectorPageSource
+        implements UpdatablePageSource
```

Comment:
If this connector supports metadata delete, then why are we using `UpdatadablePageSource`?

Reply:
Because of to prevent the error `java.lang.IllegalStateException: UpdatablePageSource not set`, this has been implemented in the `MongoPageSource` and Delete operation is supported in `UpdatablePageSource`